src/test/integration/integration-tests.py

```diff
@@ -0,0 +1,167 @@
+        valid_put = utils.put_event(valid_id, event_body)
+        validate_response(self, valid_put, 200, "events")
+
+        # Test DELETE /calendar/events/{id} with valid ID
+        valid_delete = utils.delete_event(valid_id)
+        validate_response(self, valid_delete, 204)
+
+    # Test GET /calendar/events
+    def test_get_events(self):
+        valid_get = utils.get_results("events")
+        validate_response(self, valid_get, 200)
+
+    # Test POST /calendar/events
+    def test_post_event(self):
+        # Invalid date in instance
+        good_date = event_body["data"]["attributes"]["instances"][0]["start"]
+        event_body["data"]["attributes"]["instances"][0]["start"] = "badDate"
+        bad_date = utils.post_event(event_body)
+        validate_response(self, bad_date, 400, message="Could not parse job")
+        event_body["data"]["attributes"]["instances"][0]["start"] = good_date
+
+        # Invalid campusID
+        event_body["data"]["attributes"]["campusID"] = "badCampus"
+        bad_campus = utils.post_event(event_body)
+        validate_response(self, bad_campus, 400, message="campusID")
+        event_body["data"]["attributes"]["campusID"] = None
+
+        # Invalid audienceIDs
+        validate_bad_list(self, "audienceIDs")
+
+        # Invalid countyIDs
+        validate_bad_list(self, "countyIDs")
+
+        # Invalid departmentIDs
+        validate_bad_list(self, "departmentIDs")
+
+        # Invalid eventTopicIDs
+        validate_bad_list(self, "eventTopicIDs")
+
+        # Invalid eventTypeIDs
+        validate_bad_list(self, "eventTypeIDs")
+
+    # Test PUT /calendar/events/{id}
+    def test_put_event(self):
+        # Invalid ID
+        event_body["data"]["id"] = "invalidEventID"
+        invalid_id = utils.put_event(event_body["data"]["id"], event_body)
+        validate_response(self, invalid_id, 404)
+        event_body["data"]["id"] = None
+
+        # Create test event
+        valid_event = utils.post_event(event_body)
+        valid_id = valid_event.json()["data"]["id"]
+
+        # Invalid audienceIDs
+        validate_bad_list(self, "audienceIDs", valid_id)
+        # Invalid countyIDs
+        validate_bad_list(self, "countyIDs", valid_id)
+        # Invalid departmentIDs
+        validate_bad_list(self, "departmentIDs", valid_id)
+        # Invalid eventTopicIDs
+        validate_bad_list(self, "eventTopicIDs", valid_id)
+        # Invalid eventTypeIDs
+        validate_bad_list(self, "eventTypeIDs", valid_id)
+
+        # Clean up test event
+        utils.delete_event(valid_id)
+
+    # Test GET /calendar/event-types and GET /calendar/event-types/{id}
+    def test_event_types(self):
+        test_get(self, "event-types")
+
+    # Test GET /calendar/event-topics and GET /calendar/event-topics/{id}
+    def test_event_topics(self):
+        test_get(self, "event-topics")
+
+    # Test GET /calendar/locations and GET /calendar/locations/{id}
+    def test_locations(self):
```

Comment:
This endpoint (locations) as well as campuses and departments have pagination links. Could you add tests to make sure the pagination links work as expected?